pubsublite/manager.go

```diff
@@ -240,3 +252,93 @@ func (m *Manager) DeleteSubscription(ctx context.Context, subscription string) e
 	logger.Info("deleted pubsublite subscription")
 	return nil
 }
+
+// MonitorConsumerLag registers a callback with OpenTelemetry
+// to measure consumer group lag for the given topics.
+func (m *Manager) MonitorConsumerLag(topics ...string) (metric.Registration, error) {
+	monitoringClient, err := monitoring.NewMetricClient(context.Background(), m.cfg.MonitoringClientOptions...)
```

Comment:
[for reviewer] monitoringClient is initialized in a closure and no Close is called when shutting down, since the lifecycle is controlled by caller. Practically I don't think it matters but open to suggestions.